### PR TITLE
feat(isolated-positions): Scaffold Unoponed Isolated Position cards

### PR DESCRIPTION
**Pull request by jaredvu**
https://github.com/dydxprotocol/v4-web/assets/13111994/11155585-a954-4714-a322-dae455f17123

Scaffold `Unopened Isolated Positions` section in the `<HorizontalPanel>`. This will help to navigate users to their orders on Isolated Markets before the position has been made available.

In order to test add `?isolatedMargin=1` query param

---

## Views

* New: `<UnopenedIsolatedPositions>`
  * Smart container that will eventually take Abacus state to populate a carousel of `PotentialPositionCards`.

* `<HorizontalPanel>`
  * Add `slotBottom` to display component outside of the `HorizontalPanel`
  * Updated `Styled` Record
  * Note: `styled` method from `styled-components` does not persist generic types. Had to cast `as CollapsibleTabs<InfoSection>` to get typing to work.

## Components

* New: `<PotentialPositionCard>`
  * component to display the market and margin put forth each Isolated Market

* `<CollapsibleTabs>`
  * Fix typing in `setTab` prop.

---

### CODE REVIEW

File: src/pages/trade/UnopenedIsolatedPositions.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    `}
+  `,
+  Cards: styled.div`
+    ${layoutMixins.flexWrap}
+    gap: 1rem;
+    scroll-snap-align: none;
+    padding: 0 1rem 1rem;
+  `,
+} satisfies Record<string, AnyStyledComponent>;
```
Comment (moo-onthelawn) [inline src/pages/trade/UnopenedIsolatedPositions.tsx:69]: omg the new way of typing Styled!!

File: src/pages/trade/HorizontalPanel.tsx [hunk 2]
```diff
@@ -90,6 +92,16 @@ export const HorizontalPanel = ({ isOpen = true, setIsOpen }: ElementProps) => {
...
 
+  const onViewOrders = useCallback((market: string) => {
+    navigate(`${AppRoute.Trade}/${market}`, {
+      state: {
+        from: AppRoute.Trade,
+      },
+    });
+    setView(PanelView.CurrentMarket);
+    setTab(InfoSection.Orders);
+  }, []);
```
Comment (moo-onthelawn) [inline src/pages/trade/HorizontalPanel.tsx:103]: 🙏

**Review by moo-onthelawn — APPROVED** [review]
slight nits: 

1/ I feel like if we do want to keep this blurred background behind the title of Isolated Margins, it should extend across the whole row (not just the length of the title)

Screenshot 2024-04-03 at 3 25 53 PM

2/ I wonder if it'd look slightly cleaner if we had the position cards expand horizontally such that the cards rows filled up the space available? (unsure how tricky this is - I guess we'd have to do some sort of calculation to determine number of cards before growing, etc)

File: src/pages/trade/UnopenedIsolatedPositions.tsx (hunk 1 above)
Comment (jaredvu) [reply src/pages/trade/UnopenedIsolatedPositions.tsx:69]: The `AnyStyledComponent` actually affects components that you pass to `styled(...)`, so maybe we don't even need to add the `satisfies` in the future.

**Comment by jaredvu** [comment]
re:
- blur effect: If we want to do this I can make the button span entire width, don't think it is important as of yet but good suggestion.
- dynamic width: I can play around with this further as well, maybe set a min width and let width maximize itself.

**Pull request merged**